### PR TITLE
[SYCL][Driver] Fix issue that sycl is not able to print internal defi…

### PR DESCRIPTION
**Pull request by haonanya**
clang++ -fsycl -dM -E -x c++ /dev/null is not able to print internal defines, the patch aims to fix this issue. It will emit the output to terminal as well as produce the bundled .ii file.
Signed-off-by: haonanya <haonan.yang@intel.com>

### CODE REVIEW

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -5945,6 +5952,9 @@ InputInfo Driver::BuildJobsForActionNoCache(
                                              AtTopLevel, MultipleArchs,
                                              OffloadingPrefix),
                        BaseInput);
+    if (JobForPreprocessToStdout) {
+      ResultForPreprocessToStdout = InputInfo(A, "-", BaseInput);
+    }
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:5957]:
```suggestion
    if (JobForPreprocessToStdout)
      ResultForPreprocessToStdout = InputInfo(A, "-", BaseInput);
```

Comment (haonanya) [reply clang/lib/Driver/Driver.cpp]: Modified the patch  and all pre-check-in pass already.

**Comment by haonanya** [comment]
@mdtoguchi . Modified the patch and all pre-check-in pass already.

**Comment by haonanya** [comment]
@mdtoguchi . Modified the patch and all pre-check-in pass already.
Thanks very much.

**Pull request merged**